### PR TITLE
experimentation with debugdump

### PR DESCRIPTION
**Pull request by Vrixyz**
# Objective

- I appreciate graphs from [`bevy_mod_debugdump`](https://github.com/jakobhellermann/bevy_mod_debugdump), this PR merely starts a discussion on adding some "first party" interest to such graphs.
- it can be useful to compare with bevy_rapier: https://github.com/dimforge/bevy_rapier/pull/576

## Solution

- Add an example to generate those

Run with: `cargo run --example debugdump_3d > avian.dot  && dot -Tsvg avian.dot > avian.svg`

## Output for 3d

<details><summary>PhysicsSchedule</summary>
<p>

dump_physics

</p>
</details> 

<details><summary>SubstepSchedule</summary>
<p>

dump_substeps

</p>
</details> 

<details><summary>FixedPostUpdate</summary>
<p>

dump_fixed_update

</p>
</details> 

<details><summary>Update</summary>
<p>

dump_update

</p>
</details> 

---

## Changelog

- Add an example to generate systems ordering graphs.

### CODE REVIEW

File: crates/avian3d/examples/debugdump_3d.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use avian3d::prelude::*;
+use bevy::prelude::*;
+
+fn main() {
+    let mut app = App::new();
+
+    app.add_plugins((PhysicsPlugins::default(), PhysicsDebugPlugin::default()));
+    // Schedules of interest:
+    // - PhysicsSchedule
+    // - SubstepSchedule
```
Comment (Vrixyz) [inline crates/avian3d/examples/debugdump_3d.rs:14]: Attempting to make a graph for this schedule results in an error, I'll open an upstream issue.

File: crates/avian2d/examples/debugdump_2d.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//! Run with:
+//! `cargo run --example debugdump_2d > dump.dot  && dot -Tsvg dump.dot > dump.svg`
+use avian2d::prelude::*;
+use bevy::prelude::*;
```
Comment (Jondolf) [inline crates/avian2d/examples/debugdump_2d.rs:4]:
```suggestion
//! Run with:
//! `cargo run --example debugdump_2d > dump.dot  && dot -Tsvg dump.dot > dump.svg`

use avian2d::prelude::*;
use bevy::prelude::*;
```

File: crates/avian3d/Cargo.toml [hunk 3]
```diff
@@ -82,6 +82,7 @@ bevy_math = { version = "0.14", features = ["approx"] }
 approx = "0.5"
 criterion = { version = "0.5", features = ["html_reports"] }
 insta = "1.0"
+bevy_mod_debugdump = "0.11"
 
 
 [[example]]
```
Comment (Jondolf) [inline crates/avian3d/Cargo.toml:88]:
```suggestion
bevy_mod_debugdump = "0.11"

[[example]]
```

**Review by Jondolf — APPROVED** [review]
Thanks! This is really useful. The `SubstepSchedule` issue is unfortunate, but not a blocker here imo.

**Pull request merged**